### DISCUSSION TITLE
Serilog.Sinks.Seq How buffer files will be managed in different scenarios by serilog.seq?

### DISCUSSION THREAD

**Discussion by Roman201020** [datalust/seq-tickets#1583] [Help]
Hello, I have some problems with understanding how the buffer files works in serilog seq. Any help/explanation is appreciated. So here are my questions:

1. I've discovered from this discussion #1572 , that only two last created buffer files exists until third one will be created, after that the oldest will be removed. Am I right?
2. When seq server is unreachable, I assume that buffer files will be accumulated, and the "2 last buffer files exists only" rule wont work in this scenario, until the next successful connection with the server. What is the limit of these accumulated files count? For example server is inactive for three months. In this case, should we have about buffer 180 files after these 3 months (in my case they rollback every day)?
3. Am I right, that when the size of the buffer file will be reached(I have set buffer size in serilog.seq settings), this file will be overwritten, i.e. file will be created from zero?
4. Is seq capable of handling 2000 users writing logs simultaneously using the same api key?

**Comment by nblumhardt** [comment]
Hi Roman,

1. Yes, the two most recent files are retained even when they've been shipped to Seq
2. Yes, that's correct, unless you've set a buffer size limit through `bufferSizeLimitBytes`, in which case undelivered files (apart from the most recent two) will be deleted once the size limit is exceeded
3. Files won't be overwritten in the case of the buffer size limit being exceeded; the most recent two files will continue rolling, the limit only applies to the older files (not the most recent two)

Checking the source at https://github.com/datalust/serilog-sinks-seq/tree/dev/src/Serilog.Sinks.Seq/Sinks/Seq/Durable will provide more precision if you're interested in corner cases etc.

4. This depends on the hardware, size of events, and so-on; if an ingestion burst occurs and writes stall (i.e. hardware is insufficient), Seq will generally cancel queued writes and require clients to retry, which most do

Testing with a simulation of you load and target hardware is really the only way to be certain that what you've provisioned is sufficient. High-load systems do generally require some ramp-up and tuning to get the best results. Be aware also that retained data volume is normally a source of bottlenecks well before ingestion is (another thing to test/tune for best results).

Hope this helps,
Nick
